app/src/main/java/com/fabulouszanna/pokedex/model/pokemontypes/PokemonType.kt

```diff
@@ -0,0 +1,81 @@
+package com.fabulouszanna.pokedex.model.pokemontypes
+
+sealed class PokemonType(
+    val weaknesses: List<String>,
+    val resistances: List<String> = emptyList(),
+    val immunities: List<String> = emptyList()
+) {
+    object Bug :
+        PokemonType(listOf("Fire", "Flying", "Rock"), listOf("Grass", "Fighting", "Ground"))
```

Comment:
It's an example where named arguments would prove beneficial for any reader that is fresh to the code.